linkerd/app/inbound/src/direct.rs

```diff
@@ -170,12 +171,22 @@ where
                 .check_new_service::<HttpClientInfo, FwdIo<I>>()
                 .into_inner(),
         )
+        // Use ALPN to determine whether a transport header should be read.
+        //
         // When the transport header is not present, perform HTTP detection to
         // support legacy gateway clients.
         //
-        // TODO: Stop supporting headerless connections after we have at least
-        // one stable release out with transport header support.
-        .push(svc::UnwrapOr::layer(
+        // TODO: Remove this after we have at least one stable release out
+        // with transport header support.
```

Comment:
shouldn't the comment on removing the legacy suppport for HTTP detection be on the `push_switch` rather than on the `NewTransportHeaderServer`?